### ISSUE TITLE
[Meta] E2E: look into running tests on Atomic sites

### ISSUE THREAD

**Issue by cometgrrl** [Automattic/wp-calypso#65906]
After doing some debugging on this (p1662031127146839-slack-CQD1HH4MA), I got to the point where I think we can divide those issues into two groups:
- Authentication redirect loop `flaky` failures,
- Other failures 😅 

The flaky authentication redirect loop seems to be related to the fact that we're running tests in parallel because it doesn't happen when running the tests concurrently. I don't know the root cause yet, but the nature of the redirect suggests that either auth cookies are being rejected or not added at all for some reason. For more details on this, please see p1662031127146839-slack-CQD1HH4MA.

The following (Other) failures happen on the `Gutenberg atomic E2E tests production (desktop)` [job](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_desktop/8645226?buildTab=tests) and are unrelated to the auth loop. I've created a checklist for them so that this issue can be used for tracking the fixes. The jobs for `mobile` viewport and Gutenberg `edge` sites can be addressed in a follow-up issue.

- [x] **View post as gutenbergAtomicSiteUser**
   specs/editor/editor__schedule.ts: Editor: Schedule: Schedule: future
   Fix: https://github.com/Automattic/wp-calypso/pull/67124

- [x] **Post is no longer visible**
   specs/editor/editor__post-advanced-flow.ts: Editor: Advanced Post Flow: Revert post to draft
   Fix: https://github.com/Automattic/wp-calypso/pull/67964

- [x] **upload image file using Calypso media modal** 
   specs/blocks/blocks__media.ts: Blocks: Media (Upload): Populate post with media blocks: Image block
   Fix: https://github.com/Automattic/wp-calypso/pull/68478

- [x] **upload audio file**
   specs/blocks/blocks__media.ts: Blocks: Media (Upload): Populate post with media blocks: Audio block
   Fix: https://github.com/Automattic/wp-calypso/pull/68478

- [x] **upload audio file**
   specs/blocks/blocks__media.ts: Blocks: Media (Upload): Populate post with media blocks: File block
   Fix: https://github.com/Automattic/wp-calypso/pull/68478

- [x] **Publish and visit post**
  specs/blocks/blocks__media.ts: Blocks: Media (Upload): Populate post with media blocks
   Fix: https://github.com/Automattic/wp-calypso/pull/68478

- [x] **Post metadata is found in published post** 
   specs/editor/editor__post-basic-flow.ts: Editor: Basic Post Flow: Publish
   Fix: https://github.com/Automattic/wp-calypso/pull/68651

- [x] **Block pattern count should be greater than default** 
   specs/infrastructure/infrastructure__experimental-gutenberg.ts: Gutenberg: Experimental Features
   Fix: https://github.com/Automattic/wp-calypso/pull/68662

- [x] Post a comment 
   specs/published-content/likes__comment.ts: Likes (Comment)
   Fix: https://github.com/Automattic/wp-calypso/pull/69180

- [x] Like the comment
   specs/published-content/likes__comment.ts: Likes (Comment)
   Fix: https://github.com/Automattic/wp-calypso/pull/69180

- [x] Like the comment
   specs/published-content/likes__comment.ts: Likes (Comment)
   Fix: https://github.com/Automattic/wp-calypso/pull/69180

- [x] Revision 1 displays expected diff 
   specs/editor/editor__revisions.ts: Editor: Revisions
   Fix: https://github.com/Automattic/wp-calypso/pull/69345

- [x] Restore revision 1
   specs/editor/editor__revisions.ts: Editor: Revisions
   Fix: https://github.com/Automattic/wp-calypso/pull/69345

- [x] Editor canvas loads
  specs/fse/fse__temp-smoke-test.ts: Editor: Basic Post Flow
  Fix: https://github.com/Automattic/wp-calypso/pull/69295

- [x] Navigate to Appearance > Widgets 
  specs/appearance/widgets__legacy-visibility.ts: Widgets
  ⚠️ These tests will be skipped (p1666700231937719-slack-C7YPUHBB2)
  Fix: https://github.com/Automattic/wp-calypso/pull/69571

- [x] Return to Calypso dashboard 
  specs/editor/editor__navbar.ts: Editor: Navbar
  Fix: https://github.com/Automattic/wp-calypso/pull/69422

- [x] There are no block warnings or errors in the editor 
  specs/blocks/blocks__jetpack-grow.ts: Blocks: Jetpack Grow: Add and configure blocks in the editor: Premium Content
  Additional info: https://github.com/Automattic/wp-calypso/issues/69622
  Context: p1666794081067859/1666793696.801109-slack-C1A1EKDGQ
  Fix: https://github.com/Automattic/wp-calypso/pull/69509

- [x] Configure the block
  specs/blocks/blocks__jetpack-earn.ts: Blocks: Jetpack Earn: Add and configure blocks in the editor: OpenTable
  Fix: https://github.com/Automattic/wp-calypso/pull/69556

- [x] There are no block warnings or errors in the editor
  specs/blocks/blocks__jetpack-earn.ts: Blocks: Jetpack Earn: Add and configure blocks in the editor: Payment Button
  Additional info: https://github.com/Automattic/wp-calypso/issues/69622
  Fix: https://github.com/Automattic/wp-calypso/pull/69556

- [x] There are no block warnings or errors in the editor 
  specs/blocks/blocks__jetpack-earn.ts: Blocks: Jetpack Earn: Add and configure blocks in the editor: Payment Button
  Additional info: https://github.com/Automattic/wp-calypso/issues/69622
  Fix: https://github.com/Automattic/wp-calypso/pull/69556 

- [x] Add the block from the sidebar 
  specs/blocks/blocks__etk.ts: Blocks: ETK: Add and configure blocks in the editor: Post Carousel
  Fix: https://github.com/Automattic/wp-calypso/pull/69620

- [x] Add the block from the sidebar
  specs/blocks/blocks__etk.ts: Blocks: ETK: Add and configure blocks in the editor: Timeline
  Fix: https://github.com/Automattic/wp-calypso/pull/69620

- [x] Configure the block
  specs/blocks/blocks__etk.ts: Blocks: ETK: Add and configure blocks in the editor: Timeline
  Fix: https://github.com/Automattic/wp-calypso/pull/69620

- [x] View Private page as 'defaultUser' 
  specs/editor/editor__privacy-private.ts: Editor: Privacy (Private): Validate page content
  Fix: https://github.com/Automattic/wp-calypso/pull/69425

- [x] View Private page as a public user that is not logged in.
  specs/editor/editor__privacy-private.ts: Editor: Privacy (Private): Validate page content
  Fix: https://github.com/Automattic/wp-calypso/pull/69425

- [x] View Private page as publishing user
  specs/editor/editor__privacy-private.ts: Editor: Privacy (Private): Validate page content 
  Fix: https://github.com/Automattic/wp-calypso/pull/69425

- [x] Expected content is in published post
  specs/blocks/blocks__jetpack-other.ts: Blocks: Other Jetpack Blocks: Validating blocks in published post.: Slideshow
  Fix: https://github.com/Automattic/wp-calypso/pull/69745

---

Related: #62466

**Comment by WunderBart** [comment]
@cometgrrl I've updated the description with my recent findings - hope it's a proper issue for that. I will also look into those failures as a part of the Gutenberg upgrade process. /cc @worldomonation

**Comment by worldomonation** [comment]
@WunderBart Thanks for the ping and great to see you're making progress on this one.

I should have more bandwidth in the coming weeks as Dan returns from parental leave so if you need it, I'd be glad to help.

**Comment by dpasque** [comment]
Hey @WunderBart @worldomonation! I'm looking for a couple of E2E fixes to get back into the Calypso dev flow after my parental leave. I don't want to duplicate any work you have in flight though!

Have either of you started yet on the ETK block fixes? I can peel those off and knock them out. Just let me know if you have that (or any other of these) in flight so I don't step on your toes!

**Comment by WunderBart** [comment]
> Hey @WunderBart @worldomonation! I'm looking for a couple of E2E fixes to get back into the Calypso dev flow after my parental leave. I don't want to duplicate any work you have in flight though!
> 
> Have either of you started yet on the ETK block fixes? I can peel those off and knock them out. Just let me know if you have that (or any other of these) in flight so I don't step on your toes!

Welcome back, @dpasque! 🙌 🎉 I haven't started working on anything else than the ones on that list above that have a "Fix" PR attached. To make it clear who's working on what, let's add our GH handle to the ones we pick from now on, e.g.:

> - [ ] Post a comment 
>    specs/published-content/likes__comment.ts: Likes (Comment)
>    Assignee: @WunderBart 



 Sounds good? I'll start! 😄

**Comment by WunderBart** [comment]
Nice work with all the remaining fixes, @karenroldan! 🙌 I've run the Atomic suite against trunk to check if we're all set and found [one more failure](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_desktop/8924608?buildTab=tests&status=failed) - would you like to pick this one or should I? (I've already added it to the tracking list above)

**Comment by karenroldan** [comment]
Thanks, @WunderBart! Great work! 

> I've run the Atomic suite against trunk to check if we're all set and found one more failure - would you like to pick this one or should I?

Good to see there's only one left 🥳 I can pick it up!

**Comment by karenroldan** [comment]
Ran the Atomic tests in the trunk and looks like we are all set and done! The [build](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_desktop/8942606) is now green 💚

**Comment by WunderBart** [comment]
Here's a dedicated issue for the auth redirect loop: https://github.com/Automattic/wp-calypso/issues/69865

~Also, I realized that [the mobile part](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_mobile?branch=%3Cdefault%3E&buildTypeTab=overview&mode=builds) has still [quite a few failures](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_mobile/8959845?buildTab=tests&status=failed). Should we handle them within this issue or create a new one?~

edit: There's actually [just one](https://teamcity.a8c.com/buildConfiguration/calypso_WPComTests_gutenberg_atomic_production_mobile/8961019?buildTab=tests&status=failed&expandedTest=build%3A%28id%3A8961019%29%2Cid%3A1287#%2Fscreenshots) failing test on the mobile viewport! 🙌 There was the AMP plugin enabled that was breaking all the tests so we've decided to disable it. There doesn't seem to be much value in testing with AMP enabled at this point. If we ever decide to do it though, it should probably be a dedicated site/job because of the vast UI differences. /cc @fullofcaffeine

**Comment by worldomonation** [comment]
Closing as done.
